File: src/main/java/com/procurement/clarification/model/entity/EnquiryEntity.java
package com.procurement.clarification.model.entity;

import lombok.Getter;
import lombok.Setter;
import org.springframework.data.cassandra.core.cql.PrimaryKeyType;
import org.springframework.data.cassandra.core.mapping.Column;
import org.springframework.data.cassandra.core.mapping.PrimaryKeyColumn;
import org.springframework.data.cassandra.core.mapping.Table;

@Table("clarification_enquiry")
@Getter
@Setter
public class EnquiryEntity {
    @PrimaryKeyColumn(name = "cp_id", type = PrimaryKeyType.PARTITIONED)
    private String cpId;

    @PrimaryKeyColumn(name = "stage", type = PrimaryKeyType.CLUSTERED)
    private String stage;

    @PrimaryKeyColumn(value = "token_entity", type = PrimaryKeyType.CLUSTERED)
    private String token;

    @PrimaryKeyColumn(name = "owner")
    private String owner;

    @Column(value = "json_data")
    private String jsonData;

    @Column(value = "is_answered")
    private Boolean isAnswered;
}

File: src/main/kotlin/com/procurement/clarification/application/repository/enquiry/model/EnquiryEntity.kt
package com.procurement.clarification.application.repository.enquiry.model

import com.procurement.clarification.domain.model.Cpid
import com.procurement.clarification.domain.model.Ocid
import com.procurement.clarification.domain.model.token.Token

data class EnquiryEntity(
    val cpid: Cpid,
    val ocid: Ocid,
    val token: Token,
    val isAnswered: Boolean,
    val jsonData: String
)
